### PR TITLE
Plumb through real case field names instead of generating fake ones

### PR DESCRIPTION
**Pull request by anchann**
Implemented as per @inosik's guidance in #2017.

### CODE REVIEW

**Comment by inosik** [comment]
Can you add one or two tests, please?

**Comment by anchann** [comment]
@inosik ah, yes, sorry for the lack of discipline. Updated the existing union type reflection unit test. By the way, there were a couple unit tests failing that are unrelated to my changes.

File: tests/Main/ReflectionTests.fs [hunk 1]
```diff
@@ -172,8 +172,8 @@ type TestRecord = {
 }
 
 type TestUnion =
-    | StringCase of String: string * string
-    | IntCase of Int: int
+    | StringCase of SomeString: string * string
```
Comment (anchann) [inline tests/Main/ReflectionTests.fs:175]: I didn't want to risk some coincidental equality of the field name and the (pascal cased version of?) the type, so made the field names clearly distinct.

File: tests/Main/ReflectionTests.fs [hunk 2]
```diff
@@ -364,7 +364,10 @@ let reflectionTests = [
             let types =
                 info.GetFields()
                 |> Array.map (fun field -> field.PropertyType)
-            info.Tag, info.Name, types, values)
+            let names =
```
Comment (anchann) [inline tests/Main/ReflectionTests.fs:367]: A bit duplicative, but I figured it doesn't matter in a unit test, not worth trying to change the test structure for it.

**Comment by MangelMaxime** [comment]
@anchann If the failing tests are about Date/DateTime then it is "normal". 

These tests kinds of work only if your laptop is configured on the UK timezone or something like that. We hardcoded the expected string/timezone because it was too much of an effort at the time to make the test adapt to the user machine configuration. And also, it could introduce bugs in our tests...

**Comment by anchann** [comment]
@MangelMaxime Ah, got it. Yes, it's the two DateTime/DateTimeOffset related ones.

File: tests/Main/ReflectionTests.fs (hunk 1 above)
Comment (inosik) [reply tests/Main/ReflectionTests.fs:175]: LGTM. This code is just for testing purposes.

File: tests/Main/ReflectionTests.fs [hunk 3]
```diff
@@ -27,6 +27,11 @@ type TestType3 = class end
 type TestType4 = class end
 type TestType5 = class end
 
+type TestType6 =
+    | CaseA of string
+    | CaseB of SomeField: string
+    | CaseC of string * SomeField: string
+
```
Comment (inosik) [inline tests/Main/ReflectionTests.fs]: Can this be removed? This type doesn't seem to be used.

**Comment by MangelMaxime** [comment]
Thank you for the PR @anchann

**Comment by MangelMaxime** [comment]
Thank you too @inosik for helping implementing and reviewing it

**Comment by inosik** [comment]
No problem, glad I could help 👍

**Comment by anchann** [comment]
Thanks for the review, and thank you @inosik for the implementation guidance, it made for a smooth onboarding onto a new code base!

**Pull request merged**